server/plugin/api_test.go

```diff
@@ -121,6 +121,49 @@ func TestServeFile(t *testing.T) {
 	}
 }
 
+func TestHandlePluginConfiguration(t *testing.T) {
+	for name, test := range map[string]struct {
+		SetupAPI           func(*plugintest.API) *plugintest.API
+		ExpectedStatusCode int
+		ShouldError        bool
+	}{
+		"all fine": {
+			SetupAPI:           func(api *plugintest.API) *plugintest.API { return api },
+			ExpectedStatusCode: http.StatusOK,
+			ShouldError:        false,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			api := test.SetupAPI(&plugintest.API{})
+			api.On("LogDebug", GetMockArgumentsWithType("string", 7)...).Return()
+			defer api.AssertExpectations(t)
+			p := setupTestPlugin(t, api, &mockstore.Store{})
+
+			w := httptest.NewRecorder()
+			r := httptest.NewRequest(http.MethodGet, "/api/v1/configuration", nil)
+			r.Header.Add("Mattermost-User-ID", model.NewId())
+			p.ServeHTTP(nil, w, r)
+
+			result := w.Result()
+			require.NotNil(t, result)
```

Comment:
Nit: Please close the body
```suggestion
			require.NotNil(t, result)
			defer result.Body.Close()
```